drift_detector.go

```diff
@@ -200,6 +201,32 @@ func (detector *Detector) validateApps(wg *sync.WaitGroup) {
 	successfulAppChecks.Inc()
 }
 
+func (detector *Detector) validateAppSSH(wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	var appSSHViolations []string
+
+	if !detector.cache.Apps.Valid {
+		detector.logger.Warn("invalid app cache detected. skipping ssh check.")
+		failedAppSSHChecks.Inc()
+		return
+	}
+
+	for name, expectedApp := range detector.config.Apps {
+		// only mark violations if the app was found to be deployed AND "should ssh be disabled?" == "was ssh enabled?"
+		if enabled, ok := detector.cache.Apps.sshMap[name]; ok && expectedApp.SSHDisabled == enabled {
```

Comment:
I don't like this syntax. Is this idiomatic go?

Reply:
Yes. Map lookups return (val, ok) where ok is "did the map actually contain the key". Very similar to how go expects functions to often return (value, error). See more on maps here: https://go.dev/doc/effective_go#maps.

When combined with limiting the scope of the variables to just the `if` block where they're used (as I did here) we get that kind of syntax. I _could_ move the map lookup to it's own line, but then the variables would have a larger scope than they really needed to.